pkg/services/ngalert/api/api_ruler.go

```diff
@@ -158,9 +158,17 @@ func (srv RulerSrv) RouteGetRulesConfig(c *models.ReqContext) response.Response
 		namespaceUIDs = append(namespaceUIDs, k)
 	}
 
+	dashboardUID := c.Query("dashboard_uid")
+	panelID, err := getPanelIDFromRequest(c.Req)
+	if err != nil {
+		return ErrResp(http.StatusBadRequest, err, "invalid panel_id")
```

Comment:
probably it is how we do API, and if it is then please disregard my comment, but usually, the error response should be a little bit more verbose, and mention what exactly went wrong (invalid format, missing value, etc) as well as what needs to be done to fix it. In think particular case, we can mention that the value is supposed to be an integer.

Reply:
The exact error message is wrapped in the `err` variable so the full error message would be something like `invalid panel_id: error message from err here`.